cmd/nginx-ingress/main.go

```diff
@@ -886,3 +888,11 @@ func minVersion(min string) (v *util_version.Version) {
 
 	return minVer
 }
+
+func checkDefaultServerSecretStatErr(statErr error) (exists bool, err error) {
+	if os.IsNotExist(statErr) {
+		return false, nil
+	}
+
+	return statErr == nil, statErr
```

Comment:
Sorry for the additional comments, I don't want to be too harsh, but this doesn't look correct yet. 

if os.IsNotExist() -> is true then the file doesn't exist, however you're returning (statErr == nil), shouldn't that just be ` return true, statErr` ? In which case the code becomes:
```
if os.IsNotExist(statErr) {
    return false, nil
}
return true, statErr
```
which simplifies to:
```
return os.IsNotExist(statErr), statErr
```

I don't really understand this function, it appears to be just wrapping os.IsNotExist. 

In terms of idiomatic error handling in Go, I think the usual practice is something like:
```
   if os.IsTimeout(err) {
      ...
   }
   if os.IsNotExist(err) {
       ...
   }
   ...
   // handle valid case
   return value
```

Reply:
resolved offline